File: 张宇诚/pta/L2/L2-017 人以群分.py
N = int(input())
degrees = [int(v) for v in input().split()]
degrees.sort()
outgoing, introverted, diff = 0, 0, 0

outgoing = N // 2
introverted = N // 2

if N % 2 != 0 : 
    diff = sum(degrees[:N // 2]) - sum(degrees[N // 2 + 1:])
else : 
    diff = sum(degrees[:N // 2]) - sum(degrees[N // 2:])

if N % 2 != 0 : 
    if diff > 0 : 
        introverted += 1
    else : 
        outgoing += 1
    diff = abs(diff) + degrees[N // 2]
else : 
    diff = abs(diff)

print(f"Outgoing #: {outgoing}")
print(f"Introverted #: {introverted}")
print(f"Diff = {diff}")

File: 段传相/PTA/L2/L2-017 人以群分.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> v(n), pre(n + 1);
    for (int& x : v) {
        cin >> x;
    }

    sort(v.begin(), v.begin() + n);
    //前缀和方便后面求范围内的和
    for (int i = 1; i <= n; i++) {
        pre[i] = pre[i - 1] + v[i - 1];
    }

    int l, r, diff;
    if (n % 2 == 0) {
        l = n / 2 - 1;
        r = l + 1;
        diff = pre[n] - pre[l + 1] - pre[l + 1];
    }
    else {
        l = n / 2;
        r = l + 1;
        diff = pre[n] - pre[l] - pre[l];
    }

    cout << "Outgoing #: " << r << endl;
    cout << "Introverted #: " << n - r << endl;
    cout << "Diff = " << diff;
}